src/Primitives.hs

```diff
@@ -182,10 +182,15 @@ define hidden ctx qualifiedXObj =
     defineInTypeEnv = pure . fromRight ctx . (insertTypeBinder ctx qpath)
     defineInGlobalEnv :: Binder -> IO Context
     defineInGlobalEnv newBinder =
-      when (projectEchoC (contextProj ctx)) (putStrLn (toC All (Binder emptyMeta annXObj)))
+      when (projectEchoC (contextProj ctx) && canBeEmitted annXObj) (putStrLn (toC All (Binder emptyMeta annXObj)))
         >> case (lookupBinderInGlobalEnv ctx qpath) of
           Left _ -> pure (fromRight ctx (insertInGlobalEnv ctx qpath newBinder))
           Right oldBinder -> redefineExistingBinder oldBinder newBinder
+    canBeEmitted :: XObj -> Bool
```

Comment:
There is already the function `checkForUnresolvedSymbols` in Emit.hs, would that work instead of defining this new one? (Since that function is much more complicated, I also wonder if it could be simplified, or if this one is too simplistic...)

Reply:
Ah, I forgot about that one. It looks a little more robust than my simplistic attempt at rewriting it, so I’ll change the PR to use it!